ibind/client/ibkr_client.py

```diff
@@ -62,11 +62,44 @@ def __init__(
         if url is None:
             url = f'https://{host}:{port}{base_route}'
 
+        self._use_oauth = use_oauth
+        url = var.IBIND_OAUTH_REST_URL if self._use_oauth else var.IBIND_REST_URL
+        if url is None:
+            url = f'https://{host}:{port}{base_route}'
         self.account_id = account_id
         super().__init__(url=url, cacert=cacert, timeout=timeout, max_retries=max_retries)
 
+        if self._use_oauth:
+            self.live_session_token, self.live_session_token_expires_ms = req_live_session_token(self)
+
         self.logger.info('#################')
         self.logger.info(f'New IbkrClient(base_url={self.base_url!r}, account_id={self.account_id!r}, ssl={self.cacert!r}, timeout={self._timeout}, max_retries={self._max_retries})')
 
     def make_logger(self):
         self._logger = new_daily_rotating_file_handler('IbkrClient', os.path.join(var.LOGS_DIR, f'ibkr_client_{self.account_id}'))
+
+    def get_headers(
+            self,
+            request_method: str,
+            request_url: str
+            ):
+
+        # TODO: this second check shouldn't be hardcoded. Temporary fix for now
+        # if request is for a live session token then return None
+        if (not self._use_oauth) or request_url == 'https://api.ibkr.com/v1/api/oauth/live_session_token':
```

Comment:
We can now fix this TODO by using the `self._base_url`

Reply:
Did you mean like this?
I don't think we defined ```self._base_url```
```
# TODO: this second check shouldn't be hardcoded. Temporary fix for now
# if request is for a live session token then return None
# if (not self._use_oauth) or request_url == 'https://api.ibkr.com/v1/api/oauth/live_session_token':
if (not self._use_oauth) or request_url == self._base_url:
    return {}
```

Reply:
Did you mean like this?
I don't think we defined ```self._base_url```
```
# TODO: this second check shouldn't be hardcoded. Temporary fix for now
# if request is for a live session token then return None
# if (not self._use_oauth) or request_url == 'https://api.ibkr.com/v1/api/oauth/live_session_token':
if (not self._use_oauth) or request_url == self._base_url:
    return {}```

Reply:
Sorry, I meant `self.base_url` without the underscore prefix. 

And we do have it defined in RestClient, which IbkrClient inherits from. It's the opposite situation to the one we've discussed here: https://github.com/Voyz/ibind/pull/34#discussion_r1872792619 - here IbkrClient does have access to RestClient's variables and therefore we can use it. I know this can be confusing.

We wanna do:

```python
if (not self._use_oauth) or request_url == f'{self.base_url}oauth/live_session_token':
```